src/System.Runtime/tests/System/Reflection/AssemblyTests.cs

```diff
@@ -257,55 +265,99 @@ public static void GetManifestResourceStream()
         public static void Test_GlobalAssemblyCache()
         {
             Assert.False(typeof(AssemblyTests).Assembly.GlobalAssemblyCache);
-        }        
+        }
 
         [Fact]
         public static void Test_HostContext()
         {
             Assert.Equal(0, typeof(AssemblyTests).Assembly.HostContext);
-        }        
+        }
 
         [Fact]
         public static void Test_IsFullyTrusted()
         {
             Assert.True(typeof(AssemblyTests).Assembly.IsFullyTrusted);
-        }        
+        }
 
         [Fact]
-        public static void Test_SecurityRuleSet()
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "The full .NET framework supports SecurityRuleSet")]
+        public static void Test_SecurityRuleSet_Netcore()
         {
             Assert.Equal(SecurityRuleSet.None, typeof(AssemblyTests).Assembly.SecurityRuleSet);
-        }        
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "SecurityRuleSet is ignored in .NET core")]
+        public static void Test_SecurityRuleSet_Netfx()
+        {
+            Assert.Equal(SecurityRuleSet.Level2, typeof(AssemblyTests).Assembly.SecurityRuleSet);
+        }
 
         [Fact]
         public static void Test_LoadFile()
         {
             var assem = typeof(AssemblyTests).Assembly;
             string path = "System.Runtime.Tests.dll";
             string fullpath = Path.GetFullPath(path);
-            Assert.Throws<ArgumentNullException>("path", () => Assembly.LoadFile(null));
-            Assert.Throws<ArgumentException>(() => Assembly.LoadFile(path));
             var loadfile1 = Assembly.LoadFile(fullpath);
-            Assert.NotEqual(assem, loadfile1);
```

Comment:
yes i am getting the same behavior. You should keep the assert for .net core same as what we had before and for desktop the assert should be Assert.Equal(assem, loadfile1);
It is important to compare the assembly and not just the display name